rubix-spi/src/main/java/com/qubole/rubix/spi/CacheUtil.java

```diff
@@ -48,16 +48,20 @@ public static void createCacheDirectories(Configuration conf) throws FileNotFoun
     final int maxDisks = CacheConfig.getCacheMaxDisks(conf);
 
     for (String dirPrefix : dirPrefixList) {
+      boolean parentDirectoryExists = false;
       for (int i = 0; i < maxDisks; ++i) {
         final String cacheParentDir = dirPrefix + i;
         log.debug("Checking for " + cacheParentDir);
 
         if (exists(cacheParentDir)) {
+          parentDirectoryExists = true;
           final String cacheDirPath = cacheParentDir + "/" + CacheConfig.getCacheDataDirSuffix(conf);
           createCacheDirectory(cacheDirPath);
         }
         else {
-          throw new FileNotFoundException(String.format("Cache parent directory %s does not exist", cacheParentDir));
+          if (!parentDirectoryExists) {
```

Comment:
1. Shouldnt this be outside the inner loop?

2. Better would be move this out of outer loop and throw exception only if none of the dirPrefix+i directories exist.

Reply:
Yes we can move it out. I was confused with the multiple values of dirPrefix. My thought process was we should be able to create atleast one directory with each of the dirPrefix(s).

Reply:
Even for that this needs to be outside inner loop.

Anyhow, It would be better to completely fails things only if we cannot find any place to cache so keeping it outside the outerloop.